Update docs for 3.0 release (#543)
Esri/dynamic-situational-awareness-qt: docs/README.md

Reviewer: ```suggestion
- Uses local data formats supported by the ArcGIS Native Maps SDKs, such as Geospatial PDF, Shapefile, GeoTIFF, DTED, NITF, Mobile Mosaic Dataset, GeoPackage, and Mobile Scene Packages
```

Reviewer: ```suggestion
[Tiled Basemap group on ArcGIS Online]: https://www.arcgis.com/home/group.html?id=3a890be7a4b046c7840dc4a0446c5b31#overview

[Download Map tool in ArcGIS Pro]: https://doc.esri.com/en/arcgis-pro/latest/help/projects/take-a-map-offline.html
```

Reviewer: ```suggestion
- Decide which basemaps you want to provision from among those that come with this DSA app.
- Download basemaps for offline use from [Tiled Basemap group on ArcGIS Online].
- Provision your app with basemaps of your own design.
- Use the [Download Map tool in ArcGIS Pro] to create tile packages from tile layers and imagery layers
```

Reviewer: ```suggestion
[Introduction](#dynamic-situational-awareness-dsa) | [Overview of tools and settings](#overview-of-tools-and-settings) | [Workflows and best practices](#workflows-and-best-practices) | [App architecture](#app-architecture)     
```

Reviewer: ```suggestion
# Overview of tools and settings
```

Reviewer: ```suggestion
Markups are serialized as individual JSON files in the folder specified by the `OperationalOverlays` parameter in the [app configuration file](#app-configuration-settings). 
```

Reviewer: ```suggestion
By default, the app configuration settings file is located at `~/ArcGIS/Runtime/Data/DSA/Default/DsaAppConfig.json`, where `~` is `%username%`/`C:/Users/<username>` on Windows, `$HOME` on Unix and MacOS, and `<app_folder>/files` on Android where `<app_folder>` is the path to the package name in your AndroidManifest.xml file (e.g. /data/data/com.esri.arcgisruntime.opensourceapps.DSA_Handheld_Qt).
```

Reviewer: ```suggestion
## Configure and persist app settings using the App Configuration Settings file
An app configuration settings file is used to set paths to data sources and persist the state of the app (settings, layers, and conditions) between sessions. When the DSA app runs, it will create a new app configuration file if one is not found in local storage. If the app configuration file is found, the app will read the values from the file when it starts. Any app settings changed at run time in the app using the [Settings panel](#settings-panel), data layers in the Overlays list, and alert conditions are persisted in the configuration file automatically when the app is closed. 
```

Reviewer: ```suggestion
The default path to the data used by DSA is `~/ArcGIS/Runtime/Data/DSA/Default`. If you wish to use your own data, copy the data files into the related folders (see below) and reference the above table to update the various paths in the app configuration settings file (`DsaAppConfig.json`). If you have not copied your own data into the Default directory, the app will prompt you with an option to download the demo DSA data package from ArcGIS Online (~450mb) and place it in the Default directory. 
```

Reviewer: ```suggestion

## Deploy app configuration data with the application

You can include the configuration data with the app so that it can be easily shared and deployed to another machine and run without additional configuration. To do this, store the app configurations management file (`DsaConfigurations.json`) in the same folder as the app executable, and use relative paths in the configurations management file and app configuration settings file(s) (`DsaAppConfig.json`). 
```

Reviewer: ```suggestion
As of version 3.0.0, the Configuration tab on the [Settings panel](#settings-panel) will allow you to download and deploy custom app configurations in zip files. The zip files for custom configurations follow the same specific folder structure as the sample data provided by Esri. The 'base' folders, such as 'OperationalData', 'BasemapData', etc, must be directly at the root folder level. The zip must also contain a valid `DsaAppConfig.json` file at the root of the archive. The following steps can be followed to ensure the zip file is packaged so DSA can unpack it properly.
```

Reviewer: ```suggestion
- From within the folder, select all the items (including `DsaAppConfig.json`) right click and select `Compress` on Mac or `Send to > Compressed (zipped) folder` on Windows.
```

Reviewer: ```suggestion
When you change settings on the `Options` tab, they are persisted in the application's configuration settings file. The `Configurations` tab will allow you to switch the active configuration (data + application settings), download a sample configuration from Esri, or download a custom configuration (see more info [in this section](#packaging-your-data-into-a-configuration-zip-file) or in the [Data Management](DataManagement.md#using-custom-configurations-on-mobile-platforms) doc. 
```

Author: you are right, the configurations are persisted in `DsaConfigurations.json` file

| `Observation`   | [Create observation report](#observation-report) | X | X | X |
| `Viewshed`      | [Show viewshed from track](#viewshed)  | X | X | X |

> When you open `Identify` from the context menu on the latest observation in a message feed, the popup stays dynamic and the timestamp updates as new feed messages are received.

Reviewer: ```suggestion
> [!NOTE]
>When you open `Identify` from the context menu on the latest observation in a message feed, the popup stays dynamic and the timestamp updates as new feed messages are received.
```

<!-- TODO - Verify the following paragraph: Some settings are persisted, but ALL of them? -->

> When you change settings on the `Options` tab, they are persisted in the application's configuration settings file. The `Configurations` tab will allow you to switch the active configuration (data + application settings), download a sample configuration from Esri, or download a custom configuration (see more info [in this section](#packaging-your-data-into-a-configuration-zip-file) or in the [Data Management](DataManagement.md#using-custom-configurations-on-mobile-platforms) doc. 

Reviewer: ```suggestion
When you change settings on the `Options` tab, they are persisted in the application's configuration settings file. The `Configurations` tab will allow you to switch the active configuration (data + application settings), download a sample configuration from Esri, or download a custom configuration (see more info [in this section](#packaging-your-data-into-a-configuration-zip-file) or in the [Data Management](DataManagement.md#using-custom-configurations-on-mobile-platforms) doc). 
```

- This app uses [DynamicEntities] to connect to and display message feeds in the app. [DynamicEntityLayer] is a core type added to the ArcGIS Native SDKs at version 200.1 to visualize real-time data from a [DynamicEntityDataSource].
- Military symbols are displayed using a [dictionary renderer].

### Dynamic entities

#### Track display

> ![](./images/dsa-track-display.png "Overlays")<br>*Track Display tool*

Reviewer: I think the screenshot could be a little smaller, to align with the size from the previous screenshots (i.e. the width of the Overlays ones).  You could try using html: 
`<img src=".images/dsa-track-display.png" width=300/>`

Author: updated

The Track Display under `Feeds` in Map tools provides feed-specific control of how dynamic observations and track lines are visualized.

- Select a feed from `Message Feeds` (for example, `Friendly Tracks - Air` or `Friendly Tracks - Land`) and apply display settings independently per feed.

> - Enable `Observations` to show incoming track observations for the selected feed.

Reviewer: ```suggestion
- Enable `Observations` to show historical track observations for the selected feed.
```

Author: updated

- Observation symbol size and color can be changed, and updates are reflected immediately on the map.
- `Track Length Amount` controls how much recent track history is shown.
- Enable `Track Line` to display a line representing track movement over time, with adjustable style options.
- Set `Track Length Amount` to `All` to show the full available track history for the current session.
- Observation and track-line display can be shown at the same time for both `Friendly Tracks - Air` and `Friendly Tracks - Land`, while preserving independent settings per feed.

#### Find

> ![](./images/dsa-find-feeds.png "Overlays")<br>*Find tool*

Reviewer: Same screenshot width issue. 
Also, I think it would be best to instead show an example of typing in some search text here with one or two results showing below it.

Author: updated

- This app uses [DynamicEntities] to connect to and display message feeds in the app. [DynamicEntityLayer] is a core type added to the ArcGIS Native SDKs at version 200.1 to visualize real-time data from a [DynamicEntityDataSource].
- Military symbols are displayed using a [dictionary renderer].

### Dynamic entities

#### Track display

<img src="./images/dsa-track-display.png" alt="Track Display" width="300" /><br>*Track Display tool*

> The Track Display under `Feeds` in Map tools provides feed-specific control of how dynamic observations and track lines are visualized.

Reviewer: ```suggestion
The `Track Display` tab under `Feeds` in Map tools provides feed-specific control of how dynamic observations and track lines are visualized.
```

- Select a feed from `Message Feeds` (for example, `Friendly Tracks - Air` or `Friendly Tracks - Land`) and apply display settings independently per feed.
- Enable `Observations` to show historical track observations for the selected feed.
- Observation symbol size and color can be changed, and updates are reflected immediately on the map.
- `Track Length Amount` controls how much recent track history is shown.
- Enable `Track Line` to display a line representing track movement over time, with adjustable style options.
- Set `Track Length Amount` to `All` to show the full available track history for the current session.
- Observation and track-line display can be shown at the same time for both `Friendly Tracks - Air` and `Friendly Tracks - Land`, while preserving independent settings per feed.

#### Find

<img src="./images/dsa-find-feeds.png" alt="Find" width="300" /><br>*Find tool*

> The Find under `Feeds` in Map tools helps you quickly locate specific tracks from dynamic feeds.

Reviewer: ```suggestion
The `Find` tab under `Feeds` in Map tools helps you quickly locate specific tracks from message feeds.
```

> - Select a feed type (for example, `Friendly Tracks - Land`) from the message feed type dropdown.

Reviewer: ```suggestion
- Select a message feed (for example, `Friendly Tracks - Land`) from the dropdown list.
```

| ![](./images/dsa-icon-clear-markup-32.png) | Clear all the lines of the current markup. |
| ![](./images/dsa-icon-share-markup-32.png) | The share control appears on the display after you start drawing a new markup. Use Share to finish your markup, give it a name and send it to all teammates. |

> Markups are serialized as individual JSON files in the folder specified by the `OperationalOverlays` parameter in the [app configuration file](#app-configuration-settings). 

Reviewer: ```suggestion
Markups are serialized as individual JSON (".markup") files in the folder specified by the `OperationalOverlays` parameter in the [app configuration file](#app-configuration-settings). 
```

For convenience, you can try out the DSA apps compiled for [Windows] and [Android], and the [local data] from ArcGIS Online. Please note that these compiled versions of the app are examples only, and should not be used in a production environment.

## Run on Windows desktop

There are two ways to run the app on a Windows desktop:

### Option 1: Full screen (fixed window size)

> In Windows Explorer, double-click `DSA_Vehicle_Qt.exe` to launch the app in full screen.

Reviewer: ```suggestion
In Windows Explorer, double-click `DSA_Vehicle_Qt.exe` to launch the app in full screen. This mode does not allow resizing the window. 
```

### Option 2: Window mode (resizable, can minimize, can maximize)

1. Open Command Prompt in Administrator mode.
2. Navigate to the app folder, for example:

```bat
cd C:\DSA\DSA_Vehicle\DSA_Vehicle
```

3. Run:

```bat

> DSA_Vehicle_Qt.exe --normal window

Reviewer: ```suggestion
DSA_Vehicle_Qt.exe --show normal
```

```

# App configuration settings

> 

Reviewer: ```suggestion

All data used by the app must be in a specific, predefined folder structure, referred to as an "app configuration".  At version 2.0, DSA was enhanced to support switching between different app configurations. There are two app settings files used by DSA to persist the location of configurations available to the app, and settings specific to each configuration:
- The Configurations Management file (`DsaConfigurations.json`) stores the paths to each app configuration that the user can choose to load, as well as a setting to indicate which configuration is currently loaded. 
- The App Configuration Settings file (`DsaAppConfig.json`) stores settings and paths to data particular to a specific app configuration. 

Supplemental information on how to migrate data used with DSA 1.0 to the current format, as well as how to create and deploy custom configurations, can be found in the [Data Management](DataManagement.md) documentation. 
```
